src/bootstrap/lib.rs

```diff
@@ -547,7 +553,7 @@ impl Build {
             Mode::Test => "-test",
             Mode::Codegen => "-rustc",
             Mode::Rustc => "-rustc",
-            Mode::ToolStd | Mode::ToolTest | Mode::ToolRustc => "-tools",
+            Mode::ToolOther | Mode::ToolAllLocalLibs => "-tools",
```

Comment:
This seems like a bad idea -- likely to cause problems: tools built with different std's will conflict and generally not work, we should have a `-bootstrap-tools` dir for them probably.

(That'll also mean updating the cleaning code, I think, since it shouldn't need to clean out that dir except on bootstrap compiler change)